internal/logger/targets.go

```diff
@@ -108,3 +111,37 @@ func AddTarget(t Target) error {
 
 	return nil
 }
+
+func cancelAllTargets() {
+	for _, tgt := range targets {
+		tgt.Cancel()
+	}
+}
+
+func initTargets(cfg Config) (tgts []Target, err error) {
+	for _, l := range cfg.HTTP {
+		if l.Enabled {
+			t := http.New(l)
+			if err = t.Init(); err != nil {
+				return tgts, err
+			}
+			tgts = append(tgts, t)
+		}
+	}
+	return tgts, err
+}
+
+// UpdateTargets swaps targets with newly loaded ones from the cfg
+func UpdateTargets(cfg Config) error {
+	updated, err := initTargets(cfg)
+	if err != nil {
+		return err
+	}
+
+	swapMu.Lock()
+	cancelAllTargets() // cancel running targets
+	targets = updated
+	atomic.StoreInt32(&nTargets, int32(len(updated)))
+	swapMu.Unlock()
```

Comment:
I has the order quite specific, updating `nTargets` as the first thing, since this is used outside the lock. 

I am not sure it matters, since only "zero" or non-zero matters. Going TO zero will cause targets to be empty, so no big deal.

Going *from* zero will make it take a bit longer, but I don't suppose that matters.

I think this will work. Just a bit less sure.

Reply:
I felt it made sense to update the `nTargets` after updating the targets. Nevertheless, have changed the code to what you suggested.